crates/compiler/parse/src/ast.rs

```diff
@@ -392,81 +382,126 @@ pub fn split_loc_exprs_around<'a>(
+        Expr::Num(_) => false,
+        Expr::NonBase10Int { .. } => false,
+        Expr::Str(_) => false,
+        Expr::SingleQuote(_) => false,
+        Expr::RecordAccess(a, _) => is_expr_suffixed(a),
+        Expr::AccessorFunction(_) => false,
+        Expr::TupleAccess(a, _) => is_expr_suffixed(a),
+        Expr::List(items) => items.iter().any(|x| is_expr_suffixed(&x.value)),
+        Expr::RecordUpdate { update, fields } => {
+            is_expr_suffixed(&update.value)
+                || fields
+                    .iter()
+                    .any(|field| is_assigned_value_suffixed(&field.value))
+        }
+        Expr::Record(items) => items
+            .iter()
+            .any(|field| is_assigned_value_suffixed(&field.value)),
+        Expr::Tuple(items) => items.iter().any(|x| is_expr_suffixed(&x.value)),
+        Expr::RecordBuilder(items) => items
+            .iter()
+            .any(|rbf| is_record_builder_field_suffixed(&rbf.value)),
+        Expr::IngestedFile(_, _) => false,
+        Expr::Underscore(_) => false,
+        Expr::Crash => false,
+        Expr::Tag(_) => false,
+        Expr::OpaqueRef(_) => false,
+        Expr::EmptyDefsFinal => false,
+        Expr::Backpassing(_, _, _) => false, // TODO: we might want to check this?
+        Expr::Expect(a, b) | Expr::Dbg(a, b) => {
+            is_expr_suffixed(&a.value) || is_expr_suffixed(&b.value)
+        }
+        Expr::LowLevelDbg(_, _, _) => todo!(),
+        Expr::UnaryOp(a, _) => is_expr_suffixed(&a.value),
+        Expr::When(a, _) => is_expr_suffixed(&a.value),
+        Expr::SpaceBefore(a, _) => is_expr_suffixed(a),
+        Expr::SpaceAfter(a, _) => is_expr_suffixed(a),
+        Expr::MalformedIdent(_, _) => false,
+        Expr::MalformedClosure => false,
+        Expr::MalformedSuffixed(_) => false,
+        Expr::PrecedenceConflict(_) => false,
+        Expr::MultipleRecordBuilders(_) => false,
+        Expr::UnappliedRecordBuilder(_) => false,
+    }
+}
+
+fn is_assigned_value_suffixed<'a>(value: &AssignedField<'a, Expr<'a>>) -> bool {
+    match value {
+        AssignedField::RequiredValue(_, _, a) | AssignedField::OptionalValue(_, _, a) => {
+            is_expr_suffixed(&a.value)
+        }
+        AssignedField::LabelOnly(_) => false,
+        AssignedField::SpaceBefore(a, _) | AssignedField::SpaceAfter(a, _) => {
+            is_assigned_value_suffixed(a)
+        }
+        AssignedField::Malformed(_) => false,
     }
 }
 
-pub fn wrap_in_task_ok<'a>(arena: &'a Bump, loc_expr: &'a Loc<Expr<'a>>) -> &'a Loc<Expr<'a>> {
-    arena.alloc(Loc::at(
-        loc_expr.region,
-        Expr::Apply(
-            arena.alloc(Loc::at(
-                loc_expr.region,
-                Expr::Var {
-                    module_name: ModuleName::TASK,
-                    ident: "ok",
-                    suffixed: 0,
-                },
-            )),
-            arena.alloc([loc_expr]),
-            CalledVia::BangSuffix,
-        ),
-    ))
```

Comment:
I removed this as I don't think it is necessary any longer for unwrapping trailing `!`. This felt like a bit of a hack at the time, but now without it all the tests pass, and the logic is much simpler.